### PR TITLE
Fix links between patterns

### PR DESCRIPTION
**Pull request by spier**
While working on creating a [book out of our patterns](https://github.com/InnerSourceCommons/InnerSourcePatterns/issues/173) I noticed that some reference from one pattern to another pattern were broken. (side note: Maybe we can call these inter-pattern links? :))

That issue was likely introduced when we moved around the patterns into the new maturity structure with the 3 subfolders below `patterns`.

## Fix:

- modified the link check to also check everything in the `patterns` folder.
- manually updated the inter-pattern links to point to the correct files

### CODE REVIEW

**Comment by spier** [comment]
I updated all the broken links that I could. However there are still two issues now:

## sample link in `patterns/2-structured/praise-participants.md`

This pattern provides a text-snippet sample to demo how to praise participants for their contribution.
In there we have a link to https://github.com/rcs/rcs-viewer/pull/81. That link does not exist.

Possible fixes:
- remove the link completely
- use a different example (and a working link) for praising participants

## Network timeout to http://www.fearlesschangepatterns.com

The link to http://www.fearlesschangepatterns.com times out every once in a while. Probably just a slow host.

Possible fixes:
- exclude this link from the check? (i.e. adding it to the list of ignored links for the `liche` link checker)

**Comment by lenucksi** [comment]
> I updated all the broken links that I could. However there are still two issues now:
> ## sample link in `patterns/2-structured/praise-participants.md`
> 
> This pattern provides a text-snippet sample to demo how to praise participants for their contribution.
> In there we have a link to https://github.com/rcs/rcs-viewer/pull/81. That link does not exist.
> 
> Possible fixes:
> 
>     * remove the link completely
> 
>     * use a different example (and a working link) for praising participants

Sounds like a good idea. Do we possibly have one in our own repos? Maybe @rrrutledge might know a good example.

> 
> 
> ## Network timeout to http://www.fearlesschangepatterns.com
> 
> The link to http://www.fearlesschangepatterns.com times out every once in a while. Probably just a slow host.
> 
> Possible fixes:
> 
>     * exclude this link from the check? (i.e. adding it to the list of ignored links for the `liche` link checker)

Sound good to me. Slow servers exist.

**Review by lenucksi — APPROVED** [review]
Great stuff, thanks for finding & fixing this. Added a suggestions for the open items. Happy to merge once those are solved.

**Comment by spier** [comment]
@lenucksi I ended up fixing things differently.

- updated `fearlesschangepatterns.com` from http to https (that made the link checker happy)
- added `https://github.com/rcs/rcs-viewer/pull/81` to the ignore list of the link checker, as this link is only used in an example of a **praise message** anyways, and could as well be removed.

Once this PR gets merged, all links in the files in the `patterns` directory will get checked for broken links.

**Review by lenucksi — APPROVED** [review]
LGTM

**Pull request merged**
